Dashboard/app/js/lib/views/devices/assignment-edit-views.jsx

```diff
@@ -172,18 +171,35 @@ FLOW.AssignmentEditView = FLOW.ReactComponentView.extend(
       sa.set('endDate', endDateParse);
       sa.set('startDate', startDateParse);
       sa.set('language', 'en');
-      sa.set('surveyId', FLOW.selectedControl.get('selectedSurveyGroup').get('keyId'));
+
+      if (FLOW.selectedControl.get('selectedSurveyGroup')) {
+        sa.set('surveyId', FLOW.selectedControl.get('selectedSurveyGroup').get('keyId'));
+      }
+
 
       FLOW.selectedControl.get('selectedDevices').forEach((item) => {
         devices.push(item.get('keyId'));
       });
+
+      if (!devices.length) {
+        alert('Please select a device to continue');
```

Comment:
We need to use a standardised way to do this. So for now lets not use any alerts.  Lets take this out and perhaps push the notifications/validation to the next release.

Reply:
So what do we put right now?

Reply:
We've used dialog boxes and will do the rest in this issue https://github.com/akvo/akvo-flow/issues/3221